### PR TITLE
Resolved issue regarding openign boards link from channels

### LINKED ISSUE #1373: Bug: Opening URL to board sometimes opens Dashboard instead
## Steps to reproduce the behavior

1. Go to [Focalboard community channel](https://community-daily.mattermost.com/core/channels/focalboard) (on community-daily)
2. Click on "meeting topics" in the channel header
3. Boards dashboard opens

## Expected behavior

The linked board should open

## Edition and Platform

 - Edition: Mattermost plugin
 - Version: v0.9.1 RC2
 - Browser and OS: Chrome 93 on macOS

## Additional context

This might be related to the new routes and dashboard code. It seems that opening a board url in the same tab (e.g. a relative URL) triggers a different code path / navigation.

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Resolves the issue of opening Boards via a link in Channels.

#### Ticket Link
Fixes #1373

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -51,7 +51,9 @@ if (Utils.isDesktop() && Utils.isFocalboardPlugin()) {
             return
         }
 
-        history.replace(pathName.replace((window as any).frontendBaseURL, ''))
+        if (Utils.isFocalboardPlugin()) {
```
Comment (jespino) [inline webapp/src/app.tsx:54]: This is not needed, this is going to be always true because it is checked in the outer if.

**Review by jespino — CHANGES_REQUESTED** [review]
I tiny change needed, otherwise, looks good to me.

File: webapp/src/app.tsx (hunk 1 above)
Comment (harshilsharma63) [reply webapp/src/app.tsx]: Done.

**Review by jespino — APPROVED** [review]
LGTM

**Review by hahmadia — APPROVED** [review]
Well done! Fixes the problem I had with VALIDUUID check as well.

**Pull request merged**